libfoundation/src/foundation-string.cpp

```diff
@@ -1581,31 +1581,31 @@ MC_DLLEXPORT_DEF
 codepoint_t MCStringGetCodepointAtIndex(MCStringRef self, uindex_t p_index)
 {
 	__MCAssertIsString(self);
-
-	// Calculate the code unit index for the given codepoint
-	MCRange t_codepoint_idx, t_codeunit_idx;
-    t_codepoint_idx = MCRangeMake(p_index, 1);
-    if (!MCStringMapCodepointIndices(self, t_codepoint_idx, t_codeunit_idx))
-        return 0;
  
     if (__MCStringIsIndirect(self))
         self = self -> string;
     
+    /* Allow trailing null character */
+    MCAssert(p_index <= MCStringGetLength(self));
```

Comment:
Just to be clear: it's okay for `MCStringGetCodepointAtIndex()` to retrieve the trailing nul as a codepoint? Why isn't it an error to try to read from beyond `MCString`'s explicitly-provided length?